R/stanreg-methods.R

```diff
@@ -235,32 +235,77 @@ nsamples.stanreg <- function(object, ...) {
 ranef.stanreg <- function(object, ...) {
   all_names <- if (used.optimizing(object))
     rownames(object$stan_summary) else object$stanfit@sim$fnames_oi
-  sel <- b_names(all_names)
-  ans <- object$stan_summary[sel, select_median(object$algorithm)]
-  # avoid returning the extra levels that were included
-  ans <- ans[!grepl("_NEW_", names(ans), fixed = TRUE)]
-  fl <- .flist(object)
-  levs <- lapply(fl, levels)
-  asgn <- attr(fl, "assign")
-  cnms <- .cnms(object)
-  fl <- fl
-  asgn <- asgn
-  levs <- levs
-  cnms <- cnms
-  nc <- vapply(cnms, length, 1L)
-  nb <- nc * vapply(levs, length, 1L)
-  nbseq <- rep.int(seq_along(nb), nb)
-  ml <- split(ans, nbseq)
-  for (i in seq_along(ml)) {
-    ml[[i]] <- matrix(ml[[i]], ncol = nc[i], byrow = TRUE, 
-                      dimnames = list(NULL, cnms[[i]]))
+  ans <- object$stan_summary[b_names(all_names), select_median(object$algorithm)]
+  ans <- ans[!grepl("_NEW_", names(ans), fixed = TRUE)] 
+  
+  out <- ranef_template(object)
```

Comment:
Can we call `ranef_template` first and avoid the monstrosity that is the `b_names` function?

Reply:
Yeah I’ll change that